test/jdk/javax/swing/JToolBar/ToolBarSeparatorTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4346610
+ * @key headful
+ * @summary Verifies if Adding JSeparator to JToolBar "pushes" buttons added
+ *          after separator to edge
+ * @run main ToolBarSeparatorTest
+ */
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Insets;
+import java.awt.image.BufferedImage;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JSeparator;
+import javax.swing.JToolBar;
+import javax.swing.SwingConstants;
+import javax.swing.SwingUtilities;
+
+import javax.imageio.ImageIO;
+
+public class ToolBarSeparatorTest {
+
+    private static JFrame frame;
+    private static JSeparator separator;
+    private static JToolBar toolBar;
+    private static JButton btn;
+    private static volatile Point pt;
+    private static volatile Dimension size;
+    private static volatile int sepWidth;
+    private static volatile int sepPrefWidth;
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        robot.setAutoDelay(100);
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                frame = new JFrame("Troy's ToolBarTest");
+                toolBar = new JToolBar();
+                toolBar.setMargin(new Insets(0,0,0,0));
+                btn = new JButton("button 1");
+                toolBar.add(btn);
```

Comment:
```suggestion
                toolBar.add(new JButton("button 1"));
```
The field `btn` is not used for anything else any more, it can be removed.

Reply:
ok